core/src/main/java/org/apache/carbondata/core/scan/filter/FilterExpressionProcessor.java

```diff
@@ -323,6 +323,11 @@ private FilterResolverIntf createFilterResolverTree(Expression expressionTree,
       case TRUE:
         return getFilterResolverBasedOnExpressionType(ExpressionType.TRUE, false,
             expressionTree, tableIdentifier, expressionTree);
+      case POLYGON:
+        if (expressionTree.getChildren().isEmpty()) {
```

Comment:
When getChildren will be empty ?? Please add comment

Reply:
Have refactored the code. This check is not required anymore. Have removed it.